File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        set_i = set()  
        set_j = set()  

       
        if not matrix or not matrix[0]:
            return
            
        num_rows = len(matrix)
        num_cols = len(matrix[0])

        
        for i in range(num_rows):
            for j in range(num_cols):
                if matrix[i][j] == 0:
                    set_i.add(i)
                    set_j.add(j)

        
        for k in range(num_rows):
            if k in set_i:
                for l in range(num_cols):
                    matrix[k][l] = 0

        
        for k in range(num_cols):
            if k in set_j:
                for l in range(num_rows):
                    matrix[l][k] = 0
        

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {

        HashSet<Integer> set_i = new HashSet<Integer>();
        HashSet<Integer> set_j = new HashSet<Integer>();

        for(int i=0;i<matrix.length*matrix[0].length;i++){
            int row=i/matrix[0].length;
            int col=i % matrix[0].length;
            if(matrix[row][col]==0){
                    set_i.add(row);
                    set_j.add(col);
                }

        }


        // for(int i=0;i<matrix.length;i++){
        //     for(int j=0;j<matrix[0].length;j++){
        //         if(matrix[i][j]==0){
        //             set_i.add(i);
        //             set_j.add(j);
        //         }
        //     }
        // }
        
        
        for(int k=0;k<matrix.length;k++){
            if(set_i.contains(k)){
                for(int l=0;l<matrix[k].length;l++){
                    matrix[k][l]=0;
                }
            }
        }

        for(int k=0;k<matrix[0].length;k++){
            if(set_j.contains(k)){
                for(int l=0;l<matrix.length;l++){
                    matrix[l][k]=0;
                }
            }

        }

        
    }
}